app/models/user.rb

```diff
@@ -1,2 +1,29 @@
 class User < ApplicationRecord
+  has_secure_password
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true,
+  length: {maximum: Settings.defaults.user.max_name_length}
+  validates :email, presence: true, length:
+                    {maximum: Settings.defaults.user.max_email_length},
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: {case_sensitive: false}
+  validate :birthday_must_be_within_allowed_range
+
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    email.downcase!
+  end
+
+  def birthday_must_be_within_allowed_range
+    return if birthday.blank?
+
+    max_years = Settings.defaults.user.max_age_years
+    if birthday < max_years.years.ago.to_date || birthday > Time.zone.today
+      errors.add(:birthday, I18n.t("users.birthday_year_validation", count: max_years))
```

Comment:
```suggestion
      errors.add(:birthday, :birthday_year_validation, count: max_years)
```
Chỗ này dùng key i18n chứ k dùng messages chỗ này
Đặt key birthday_year_validation này vào file i18n với format này
https://viblo.asia/p/internationalization-i18n-trong-ruby-on-rails-YWOZrxBR5Q0#:~:text=Ti%E1%BA%BFp%20theo%20cung,chu%E1%BB%97i%20qu%C3%A1%20ng%E1%BA%AFn